protocol/xmrmaker/checks.go

```diff
@@ -94,7 +94,7 @@ func checkContractSwapID(msg *message.NotifyETHLocked) error {
 func (s *swapState) checkContract(txHash ethcommon.Hash) error {
 	tx, _, err := s.ETHClient().Raw().TransactionByHash(s.ctx, txHash)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to get transaction by hash: %w", err)
```

Comment:
Would it be correct to write `failed to get newSwap transaction by hash`? Knowing which transaction wasn't found would be helpful. The function comments should probably state which txHash is being passed as well (or it could be placed into the parameter name).